tensorpipe/test/CMakeLists.txt

```diff
@@ -4,87 +4,104 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-add_executable(tensorpipe_test
-  test.cc
-  test_environment.cc
-  transport/context_test.cc
-  transport/connection_test.cc
-  transport/uv/uv_test.cc
-  transport/uv/context_test.cc
-  transport/uv/loop_test.cc
-  transport/uv/connection_test.cc
-  transport/uv/sockaddr_test.cc
-  transport/listener_test.cc
-  core/context_test.cc
-  channel/basic/basic_test.cc
-  channel/xth/xth_test.cc
-  channel/mpt/mpt_test.cc
-  channel/channel_test.cc
-  channel/channel_test_cpu.cc
-  common/system_test.cc
-  common/defs_test.cc
+# TP_SRCS is the list of source files that we need to build libtensorpipe.
+set(TP_TEST_SRCS)
+set(TP_TEST_CUDA_SRCS)
+set(TP_TEST_ROCM_SRCS)
+
+# TP_HDRS is the list of header files that we need to install.
+set(TP_TEST_HDRS)
+set(TP_TEST_CUDA_HDRS)
+set(TP_TEST_ROCM_HDRS)
```

Comment:
What is the rationale behind using variables here, as they do not seem to be used outside of adding files to targets?

Reply:
@beauby,
I made these changes thinking it will help in controlling the files to be compiled on ROCm after hipifying.
Currently the result of hipifying, creates few files(pipe.cc -> pipe_hip.cc) and folders (cuda_basic -> hip_basic). 

Checking on a implementing the cudalib class as a derived class of devicelib base class and rocmlib derived class correspondingly. If this approach is fine, may be we can skip this commit (restructuring changes to cmake files).
Let me know your thoughts.